src/Admin/AdminConsole/Controllers/ProvidersController.cs

```diff
@@ -295,6 +295,25 @@ public async Task<IActionResult> Edit(Guid id, ProviderEditModel model)
             return View(oldModel);
         }
 
+        // validate the stripe ids to prevent saving a bad one
+        if (provider.IsBillable())
+        {
+            if (!string.IsNullOrEmpty(model.GatewayCustomerId) &&
+                !await _providerBillingService.IsValidGatewayCustomerIdAsync(model.GatewayCustomerId))
+            {
+                var oldModel = await GetEditModel(id);
+                ModelState.AddModelError(nameof(model.GatewayCustomerId), $"Invalid Gateway Customer Id: {model.GatewayCustomerId}");
+                return View(oldModel);
+            }
+            if (!string.IsNullOrEmpty(model.GatewayCustomerId) &&
+                !await _providerBillingService.IsValidGatewaySubscriptionIdAsync(model.GatewaySubscriptionId))
+            {
+                var oldModel = await GetEditModel(id);
+                ModelState.AddModelError(nameof(model.GatewaySubscriptionId), $"Invalid Gateway Subscription Id: {model.GatewaySubscriptionId}");
+                return View(oldModel);
+            }
+        }
```

Comment:
(non-blocking) This is an improvement but I wonder if the interface could be further abstracted. For example, could it take the whole `model` object and validate whatever properties it needs to, and then return a result which the controller handles?

The current design requires me (in a different team) to know which properties need to be validated by Billing Team and call the appropriate interfaces one-by-one.

I am hammering this point a little but cross-team interfaces is something @amorask-bitwarden and I have been discussing semi-recently.

Again, this is non-blocking so I will leave it to you as to whether you iterate on it further.